packages/sanity/src/core/config/types.ts

```diff
@@ -175,6 +173,11 @@ export interface Tool<Options = any> {
     params: Record<string, unknown>,
     payload: unknown,
   ) => boolean | {[key: string]: boolean}
+
+  /**
+   * The type of tool – used to group tools across multiple workspaces and have a constant identifier.
+   */
+  type?: string
```

Comment:
> Is there a foreseen issue adding type to tools?

It says "used to group tools", but the Studio doesnt do that. It is not clear to me how we want to design types/apis for studio types that are not used by studio but only by <redacted>.

Reply:
The type concept is a bit fuzzy to me as well. Probably missing some context. If it's for grouping, could it be a way for the tool to declare what group it belongs to, e.g. by `group: 'edit'`? If it's for identifying it, should the property be named "id" instead? Feels like it should have been the "name" property.

Is the idea that the consuming end doesn't need to know what types of tools exists; i.e. it just groups the ones with similar types? Or could the consuming aware of what types exists and will do the grouping based on that?

- How do we guide plugin authors to chose type/group/identifier?
- Are the of which tools are appearing in the other end ever going to be important?
- How do we avoid name clashes/unintentionally grouping ?

Reply:
For future people – We've discussed this internally, it's use case is primarily to manage sanity tools we expect to turn into applications in the future, we don't really need / want third party developers to be concerned or confused with this. As such we've removed from the type signature & underscored it so it's clearly undocumented & internal to work with for now.